core/bin/zksync_server/src/node_builder.rs

```diff
@@ -296,7 +296,13 @@ impl MainNodeBuilder {
     }
 
     fn add_eigenda_proxy_layer(mut self) -> anyhow::Result<Self> {
-        self.node.add_layer(EigenDAProxyLayer::new());
+        let da_config = try_load_config!(self.configs.da_client_config);
+        match da_config {
+            DAClientConfig::EigenDA(config) => {
+                self.node.add_layer(EigenDAProxyLayer::new(config));
+            }
+            _ => {}
```

Comment:
loading a config, but not matching to any `da_config` should result in an `Error`.

Reply:
I don't think it should be an error, if you are using avail for example, you don't want the proxy layer to be added, this takes care of that.
You cannot add it, since it would later need the specific config for eigenda, which it would not have